Makefile

```diff
@@ -377,4 +377,20 @@ test-perf: local
 	$(MAKE) -e VERSION=$(VERSION) -C test/ run-perf
 
 go-generate:
-	go generate ./pkg/...
\ No newline at end of file
+	go generate ./pkg/...
+
+# requires an authenticated gh cli
+# gh: https://cli.github.com/
+# by default uses PR title as changelog body but can be overwritten like so
+# make new-changelog CHANGELOG_BODY="Changes you have made"
+new-changelog: GH_LOGIN ?= $(shell gh pr view --json author --jq .author.login 2> /dev/null)
+new-changelog: GH_PR_NUMBER ?= $(shell gh pr view --json number --jq .number 2> /dev/null)
```

Comment:
I never used `gh` cmd, but is this always reliable?

Reply:
It's reliable. It's GitHub's own CLI.

Reply:
It gets the right PR number for me.

I could add more vars here for base branch for those who work on other downstream but for folks with their fork and tanzu this should work out of the box.

Reply:
The alternative is to curl against GitHub rest API

Reply:
It's based on the current branch you checked out. GitHub will check if the branch is used as a PR against upstream.

Reply:
@kaovilai we'd need to document somewhere that users need to install the github CLI, though.

Reply:
Line 382 count?

Reply:
Oh, yeah -- that's probably good enough. And this is an optional task that's not called anywhere else, so if it's not there, it won't matter as long as this task isn't executed explicitly.